xtask/src/ci.rs

```diff
@@ -95,4 +102,77 @@ impl CiTask {
 
         fmt_res.and(check_full_res).and(clippy_res).and(sort_res)
     }
+
+    fn check_spec_links(&self) -> Result<()> {
+        let mut start_path = self.project_root.clone();
+        start_path.push("crates");
+        walk_dirs(start_path)
+    }
+}
+
+fn walk_dirs<P: AsRef<Path>>(path: P) -> Result<()> {
+    fn check_rest(rest: &[&str], path: &Path) -> Result<()> {
+        match rest {
+            [] => Ok(()),
+            [rest] => {
+                if !(rest.starts_with("r0.6.1#") || rest.starts_with("unstable#")) {
+                    Err(format!(
+                        "error: spec URL with no version number\nfrom: {}\n{}",
+                        path.display(),
+                        &rest[0..25]
+                    )
+                    .into())
+                } else {
+                    Ok(())
+                }
+            }
+            [pre, post, rest @ ..] => {
+                if pre.lines().rev().next().map_or(false, |l| l.starts_with("//!"))
+                    && !(post.starts_with("r0.6.1#") || post.starts_with("unstable#"))
+                {
+                    println!("{}", pre);
+                }
+                if rest.is_empty() {
+                    check_rest(rest, path)?;
+                }
+                Ok(())
+            }
+        }
+    }
+
+    if path.as_ref().is_dir() {
+        for entry in fs::read_dir(path)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_dir() {
+                walk_dirs(&path)?;
+            } else {
+                let content = fs::read_to_string(&path)?;
+                let split = content
+                    .split("https://matrix.org/docs/spec/client_server/")
```

Comment:
This only checks the first occurrence of `https://matrix.org/docs/spec/client_server/` in every file, right?

Reply:
No, it checks every occurance if it's in mod doc comments except in this case https://github.com/ruma/ruma/pull/735/commits/412ab8f0f699097b775088d1b50f095eef2ecc34#diff-41845fe4e4cd68d46a859e827df93ca1da66a9d62bbf6fcb3a5438cbd2a1b782R179 where there is no way to check that.

Reply:
Oh it's the iterator thing. I was thinking of `split_once`.